Game/HidingSystem.gd

```diff
@@ -24,19 +25,16 @@ func choose_hiding_spot():
     var next_possible_hiding_spots = hiding_spots.duplicate()
     next_possible_hiding_spots.erase(current_hiding_spot)
     current_hiding_spot = next_possible_hiding_spots[randi_range(0, next_possible_hiding_spots.size())-1]
+    emit_signal("new_hiding_spot", current_hiding_spot)
```

Comment:
This emits the new_hiding_spot signal, which the Cat node can listen to in order to figure out its next place to navigate to and hide.